src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/FlowHelper.groovy

```diff
@@ -50,7 +50,7 @@ class FlowHelper {
 
     def random = new Random()
     def faker = new Faker()
-    def allowedVlans = 101..4095
+    def allowedVlans = 101..4094
```

Comment:
Actually, it must be explained (or, at least, noticed) in swagger, that correct values for VLAN are in range 1..4094.

Reply:
right, but we see the change here but not in swagger. That means while it was explained in swagger in one specific way, the test has different values. I think I am worried here about those usual one-off errors. As well as consistency, because in other places (specs) it is like `Vlan must be in range [0, 4095]`. It's difficult to spot the error without the context. 
I believe a comment would make it clearer. Kind of `//Yep. 4094 is correct here`

Reply:
This is a fix after https://github.com/telstra/open-kilda/pull/4892 highlighted the problem. In fact, we had wrong VLANs range in tests for years.

I don't agree, that each change in code should be commented. 'allowedVlans' variable name is self-explaining, IMHO.

Reply:
Hm. Do I get it right, that the only places, where we see 4094 vs 4095, are this PR and git history?
If so, we have an explanation in PR and anyone can go to this PR from git history and read an explanation as well? 
I guess, the problem is solved :D

Reply:
https://github.com/telstra/open-kilda/blob/f7d69aeddec0ec2723d80f6a290eb5d2dc0a848c/src-java/base-topology/base-messaging/src/main/java/org/openkilda/messaging/Utils.java#LL114C27-L114C27
any changes required here?

Reply:
I just briefly looked at full text search, there are occurrences. For example https://github.com/telstra/open-kilda/blob/f7d69aeddec0ec2723d80f6a290eb5d2dc0a848c/src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy#L450

Reply:
> https://github.com/telstra/open-kilda/blob/f7d69aeddec0ec2723d80f6a290eb5d2dc0a848c/src-java/base-topology/base-messaging/src/main/java/org/openkilda/messaging/Utils.java#LL114C27-L114C27 any changes required here?

No, as far as I understand. VLANs 0 and 4095 have a special roles, so that they can be used by Kilda, but can not be used to create valid flows.

Reply:
this is the information needed near the constant. It might or might not be valid to use 4095

Reply:
> I just briefly looked at full text search, there are occurrences. For example
> 
> https://github.com/telstra/open-kilda/blob/f7d69aeddec0ec2723d80f6a290eb5d2dc0a848c/src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy#L450

These are used in negative tests, which verify the API input validation.

> this is the information needed near the constant. It might or might not be valid to use 4095

This is the information from RFC. I don't remember someone in spring-boot-web leave comment near every HTTP status code with quote from RFC, proving that value in sources adheres to the standard.

Reply:
I don't get your point. You leave comment even with some jokes in the code. What is going on here?
The code snippet I provided was just an example (see "for example"). There is no need to explain every occurrence of 4095 here in this thread. What is needed is to analyze the usages and make an effort to decrease the chances of  "we had wrong VLANs range in tests for years" happened again

Reply:
please just add a comment similar you wrote here to the code. For example, something like this:

```suggestion
    // VLANs 0..100 and 4095 have a special roles, so that they can be used by Kilda, but can not be used to create valid flows.
    def allowedVlans = 101..4094
```

As for refactoring, please the git issue with your proposal.